libraries/chain/combined_database.cpp

```diff
@@ -494,19 +514,16 @@ namespace eosio { namespace chain {
                         move_to_rocks = &row;
                      });
                      if (move_to_rocks) {
-                        auto buffer = std::vector<char>{};
-                        buffer.reserve(sizeof(uint64_t) + prefix_key.size() + move_to_rocks->kv_key.size());
-                        buffer.insert(std::end(buffer), prefix_key.data(), prefix_key.data() + prefix_key.size());
-                        b1::chain_kv::append_key(buffer, move_to_rocks->contract.to_uint64_t());
-                        buffer.insert(std::end(buffer), move_to_rocks->kv_key.data(),
-                                      move_to_rocks->kv_key.data() + move_to_rocks->kv_key.size());
+                        auto key = eosio::session::shared_bytes{sizeof(uint64_t) + prefix_key.size() + move_to_rocks->kv_key.size()};
```

Comment:
This code is very inefficient, move_to_rocks is created in the CHAINBASE and then is removed at the end.  I have rewritten all of this.

Reply:
Which code?  And what are the performance metrics indicating that?

Reply:
The surrounding code.  It took advantage of the db to create the move_to_rocks, then deleted it afterwards. It doesn't need to do that.  I'm changing this code to use make_shared_bytes.